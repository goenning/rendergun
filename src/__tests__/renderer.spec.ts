import * as fs from "fs";
import Renderer from "../renderer";

describe("Renderer", () => {
  const renderer = new Renderer();

  beforeAll(async () => {
    await renderer.initialize();
  });

  afterAll(async () => {
    await renderer.close();
  });

  test("can render example.org", async () => {
    jest.setTimeout(8000);

    const result = await renderer.render("https://example.org");
    expect(result.code).toBe(200);
    expect(result.body).toMatchSnapshot();
  });

  test("can render pudim.com.br (without script tag)", async () => {
    jest.setTimeout(8000);

    const result = await renderer.render("http://www.pudim.com.br");
    expect(result.code).toBe(200);
    expect(result.body).toMatchSnapshot();
  });

  test("returns 404 on an unknown page", async () => {
    jest.setTimeout(8000);

    const result = await renderer.render("http://pudim.com.br/abc");
    expect(result.code).toBe(404);
    expect(result.body).toMatchSnapshot();
  });

  test("can render a simple string", async () => {
    jest.setTimeout(8000);

    const content = fs.readFileSync("./src/__tests__/testdata/simple-string.html", "UTF-8");
    const result = await renderer.renderString("https://example.org", content);
    expect(result.code).toBe(200);
    expect(result.body).toMatchSnapshot();
  });

  test("can render javascript content", async () => {
    jest.setTimeout(8000);

    const content = fs.readFileSync("./src/__tests__/testdata/simple-javascript.html", "UTF-8");
    const result = await renderer.renderString("https://example.org", content);
    expect(result.code).toBe(200);
    expect(result.body).toMatchSnapshot();
  });

  test("can render Fider home page (react)", async () => {
    jest.setTimeout(8000);

    const content = fs.readFileSync("./src/__tests__/testdata/fider-home.html", "UTF-8");
    const result = await renderer.renderString("https://trax.fider.io/", content);
    expect(result.code).toBe(200);
    expect(result.body).toMatchSnapshot();
  });
});
