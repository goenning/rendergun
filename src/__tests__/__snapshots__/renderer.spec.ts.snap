// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer can render a page using JQuery 1`] = `
"<!DOCTYPE html><html><head>
  <title>JQuery + Bootstrap</title>
  <link rel=\\"stylesheet\\" href=\\"https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css\\" integrity=\\"sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T\\" crossorigin=\\"anonymous\\">
</head>
<body>
  
  <span id=\\"this-location\\">https://this-url-doesnt-matter.com/</span>
  <ul>
    <li>This is JQUERY!</li>
    <li>This is JQUERY!</li>
    <li>This is JQUERY!</li>
  </ul>
  
  



</body></html>"
`;

exports[`Renderer can render a simple string 1`] = `"<html><head></head><body>Hello World</body></html>"`;

exports[`Renderer can render example.org 1`] = `
"<!DOCTYPE html><html><head>
    <title>Example Domain</title>

    <meta charset=\\"utf-8\\">
    <meta http-equiv=\\"Content-type\\" content=\\"text/html; charset=utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <style type=\\"text/css\\">
    body {
        background-color: #f0f0f2;
        margin: 0;
        padding: 0;
        font-family: -apple-system, system-ui, BlinkMacSystemFont, \\"Segoe UI\\", \\"Open Sans\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;
        
    }
    div {
        width: 600px;
        margin: 5em auto;
        padding: 2em;
        background-color: #fdfdff;
        border-radius: 0.5em;
        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);
    }
    a:link, a:visited {
        color: #38488f;
        text-decoration: none;
    }
    @media (max-width: 700px) {
        div {
            margin: 0 auto;
            width: auto;
        }
    }
    </style>    
</head>

<body>
<div>
    <h1>Example Domain</h1>
    <p>This domain is for use in illustrative examples in documents. You may use this
    domain in literature without prior coordination or asking for permission.</p>
    <p><a href=\\"https://www.iana.org/domains/example\\">More information...</a></p>
</div>


</body></html>"
`;

exports[`Renderer can render javascript content 1`] = `
"<html><head></head><body>
    <div id=\\"root\\">A JavaScript Injection</div>
    
  
</body></html>"
`;

exports[`Renderer can render pudim.com.br (without script tag) 1`] = `
"<html xmlns=\\"http://www.w3.org/1999/xhtml\\"><head>
    <title>Pudim</title>
    <link rel=\\"stylesheet\\" href=\\"estilo.css\\">
</head>
<body>
<div>
    <div class=\\"container\\">
        <div class=\\"image\\">
            <img src=\\"pudim.jpg\\" alt=\\"\\">
        </div>
        <div class=\\"email\\">
            <a href=\\"mailto:pudim@pudim.com.br\\">pudim@pudim.com.br</a>
        </div>
    </div>
</div>



</body></html>"
`;

exports[`Renderer returns 404 on an unknown page 1`] = `
"<!DOCTYPE html PUBLIC \\"-//IETF//DTD HTML 2.0//EN\\"><html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL /abc was not found on this server.</p>

</body></html>"
`;
